lib/explorer/data_frame.ex

```diff
@@ -5615,14 +5620,50 @@ defmodule Explorer.DataFrame do
   Regexes and functions are also accepted in column names, as in `select/2`.
   """
   @doc type: :single
-  @spec group_by(df :: DataFrame.t(), groups_or_group :: column_names() | column_name()) ::
+  @spec group_by(
+          df :: DataFrame.t(),
+          groups_or_group :: column_names() | column_name(),
+          opts :: [stable: boolean()]
+        ) ::
           DataFrame.t()
-  def group_by(df, group) when is_column(group), do: group_by(df, [group])
+  def group_by(df, group, opts \\ [])
+  def group_by(df, group, opts) when is_column(group), do: group_by(df, [group], opts)
+
+  def group_by(df, groups, opts) do
+    opts = Keyword.validate!(opts, [:stable])
 
-  def group_by(df, groups) do
     groups = to_existing_columns(df, groups)
-    all_groups = Enum.uniq(df.groups ++ groups)
-    %{df | groups: all_groups}
+    all_groups = Enum.uniq(df.groups.columns ++ groups)
+
+    # TODO rewrite
+    stable_opt =
+      case Keyword.fetch(opts, :stable) do
+        :error ->
+          nil
+
+        {:ok, value} when is_boolean(value) ->
+          value
+
+        {:ok, value} ->
+          raise ArgumentError, "`:stable` must be `true` or `false`, found: #{inspect(value)}."
+      end
```

Comment:
@petrkozorezov How are you feeling besides this TODO?

Reply:
Forgot to remove the TODO.